lib/src/transformer/transformer.dart

```diff
@@ -68,8 +81,9 @@ class _TransformAssetWriter implements AssetWriter {
   _TransformAssetWriter(this.transform);
 
   @override
-  Future writeAsString(build.Asset asset, {Encoding encoding: UTF8}) async =>
-      transform.addOutput(_toBarbackAsset(asset));
+  Future writeAsString(build.Asset asset, {Encoding encoding: UTF8}) async {
```

Comment:
Ya actually I agree with you just because of the weirdness of running `transform.addOutput` async, if nothing else ;).